index.js

```diff
@@ -612,4 +617,18 @@ HtmlWebpackPlugin.prototype.getAssetFiles = function (assets) {
   return files;
 };
 
+/**
+ * Helper to promisify compilation.applyPluginsAsyncWaterfall that returns
+ * a function that helps to merge given plugin arguments with processed ones
+ */
+HtmlWebpackPlugin.prototype.applyPluginsAsyncWaterfall = function (compilation) {
+  var promisedApplyPluginsAsyncWaterfall = Promise.promisify(compilation.applyPluginsAsyncWaterfall, {context: compilation});
+  return function (eventName, pluginArgs) {
+    return promisedApplyPluginsAsyncWaterfall(eventName, pluginArgs)
+      .then(function (result) {
+        return _.extend(pluginArgs, result);
```

Comment:
Can you please add a warning to the compilation if the result is undefined?

"`Using webpack-html-plugin without returning a result is deprecated.`"